src/main/java/net/logstash/logback/marker/LogstashBasicMarker.java

```diff
@@ -107,6 +70,13 @@ public void add(Marker reference) {
             return;
         }
         
+        if (referenceList == null) {
+            synchronized (this) {
+                if (referenceList == null) {
+                    this.referenceList = new CopyOnWriteArrayList<>();
+                }
+            }
+        }
         referenceList.add(reference);
```

Comment:
_nit_ Referencing this field without using `this.`, and with using `this.`, in this single method is a little confusing when reviewing.  At first I thought they were different things.  Can you either remove `this.` everywhere, or add `this.` everywhere to use one approach.